cmd/tools/gendynamicconfig/main.go

```diff
@@ -0,0 +1,245 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/template"
+)
+
+type (
+	settingType struct {
+		Name   string
+		GoType string
+		Index  int
+	}
+	settingPrecedence struct {
+		Name       string
+		GoArgs     string
+		GoArgNames string
+		Index      int
+	}
+)
+
+var (
+	types = []*settingType{
+		{
+			Name:   "Bool",
+			GoType: "bool",
+		},
+		{
+			Name:   "Int",
+			GoType: "int",
+		},
+		{
+			Name:   "Float",
+			GoType: "float64",
+		},
+		{
+			Name:   "String",
+			GoType: "string",
+		},
+		{
+			Name:   "Duration",
+			GoType: "time.Duration",
+		},
+		{
+			Name:   "Map",
+			GoType: "map[string]any",
+		},
+	}
+	precedences = []*settingPrecedence{
+		{
+			Name:       "Global",
+			GoArgs:     "()",
+			GoArgNames: "()",
+		},
+		{
+			Name:       "Namespace",
+			GoArgs:     "(namespace string)",
+			GoArgNames: "(namespace)",
+		},
+		{
+			Name:       "NamespaceID",
+			GoArgs:     "(namespaceID string)",
+			GoArgNames: "(namespaceID)",
+		},
+		{
+			Name:       "TaskQueue",
+			GoArgs:     "(namespace string, taskQueue string, taskQueueType enumspb.TaskQueueType)",
+			GoArgNames: "(namespace, taskQueue, taskQueueType)",
+		},
+		{
+			Name:       "ShardID",
+			GoArgs:     "(shardID int32)",
+			GoArgNames: "(shardID)",
+		},
+		{
+			Name:       "TaskType",
+			GoArgs:     "(taskType enumsspb.TaskType)",
+			GoArgNames: "(taskType)",
+		},
+	}
+)
+
+func panicIfErr(err error) {
+	if err != nil {
+		panic(err)
```

Comment:
Do we need the panic stack trace or could we `log.Fatal(err)` instead?

Reply:
I'm fine with that